site/src/api/api.ts

```diff
@@ -411,11 +411,13 @@ export const unarchiveTemplateVersion = async (templateVersionId: string) => {
   return response.data;
 };
 
+// updateTemplateMeta will return "undefined" if there is no change to the template.
+// A noop returns a "Not Modified" (304) response.
 export const updateTemplateMeta = async (
   templateId: string,
   data: TypesGen.UpdateTemplateMeta,
-): Promise<TypesGen.Template> => {
-  const response = await axios.patch<TypesGen.Template>(
+): Promise<TypesGen.Template | undefined> => {
+  const response = await axios.patch<TypesGen.Template | undefined>(
```

Comment:
Oh wait, I just realized something – I'm worried the logic isn't quite right. `undefined` isn't a valid JSON-serializable value (though `null` is), so it shouldn't be able to make it across the API boundary in the first place

Is the condition we should actually be checking for whether `data.name` is the string literal `"undefined"`? Even in the bug issue, the previous code wasn't breaking when it was trying to access the `name` property during the navigation – it was just being fed an incorrect value. That's making me think that technically, `data` and its properties would always be defined – it's just that `name` is `"undefined"` in certain situations

This is definitely out of scope for the PR, and I haven't explored our type generation logic, so I don't know how complicated this would be, but there is a trick we can use to add the literal `"undefined"` to autocomplete without narrowing the overall type unnecessarily:
```ts
interface Template {
  readonly name: "undefined" | (string | {}); // (string | NonNullable<unknown>) also works
}
```
With that, "undefined" would show up during autocomplete, but the overall type would still be `string`

Reply:
The response is a completely empty payload, 0 bytes.

You are saying that is `null` and not `undefined`?